reasoner/controller/ConjunctionController.java

```diff
@@ -184,80 +194,115 @@ NegatedRequest asNegated() {
+                this.plan = plan;
+                Publisher<ConceptMap> leftChild;
+                if (plan.isCompoundStream()) {
+                    leftChild = spawnPlanElement(plan.asCompoundStreamPlan().ithChild(0), identifierBounds);
+                } else {
+                    leftChild = spawnPlanElement(plan.asResolvablePlan(), identifierBounds);
+                }
+                leftChild.registerSubscriber(this);
+                this.whichChild = new HashMap<>();
+                this.whichChild.put(leftChild, 0);
                 processor().context().perfCounters().compoundStreams.add(1);
             }
 
+            private Publisher<ConceptMap> spawnPlanElement(ConjunctionStreamPlan planElement, ConceptMap bounds) {
+                ConceptMap extension = filterOutputsWithExplainables(bounds, planElement.extendOutputWithVariables());
+                ConceptMap identifiers = bounds.filter(planElement.identifierVariables());
+                assert planElement.identifierVariables().size() == identifiers.concepts().size() &&  identifiers.concepts().keySet().containsAll(planElement.identifierVariables());
+                Publisher<ConceptMap> publisher;
+                if (planElement.isResolvable()) {
+                    publisher = spawnResolvableElement(planElement.asResolvablePlan(), identifiers);
+                } else if (planElement.isCompoundStream()) {
+                    publisher = spawnCompoundStream(planElement.asCompoundStreamPlan(), identifiers);
+                } else throw TypeDBException.of(ILLEGAL_STATE);
+
+                return extendWithBounds(publisher, extension);
+            }
+
             @Override
             public Either<Publisher<ConceptMap>, Set<ConceptMap>> accept(Publisher<ConceptMap> publisher,
                                                                          ConceptMap packet) {
-                ConceptMap mergedPacket = merge(initialPacket, packet);
-                if (leadingPublisher.equals(publisher)) {
-                    if (remainingPlan.size() == 0) {  // For a single item plan
-                        return Either.second(set(mergedPacket));
-                    } else {
-                        Publisher<ConceptMap> follower;  // TODO: Creation of a new publisher should be delegated to the owner of this operation
-                        if (remainingPlan.size() == 1) {
-                            follower = nextCompoundLeader(remainingPlan.get(0), mergedPacket);
-                        } else {
-                            follower = new CompoundStream(processor, remainingPlan, mergedPacket).buffer();
-                        }
-                        publisherPackets.put(follower, mergedPacket);
-                        return Either.first(follower);
-                    }
+                context().perfCounters().compoundStreamAccepts.add(1);
+                ConceptMap mergedPacket = merge(identifierBounds, packet);
```

Comment:
This line is a bit sus. I would like to see if we can just keep these identifier bounds in the extension (join) variables to handle this.